### DISCUSSION TITLE
Upserting a document and nesting an upsert of a one-to-many relation using a key that is created when the parent document is created?

### DISCUSSION THREAD

**Discussion by matteopolak** [prisma/prisma#15821] [Q&A] [answered]
Hey there!

I'm looking to upsert a `Profile` if the `User` does not have one with the `Mode` yet. Every `User` should only have one `Profile` per `Mode`, so I think creating a unique compound field with the `userId` and `mode` is correct?

When upserting a user, I need to also upsert a `Profile` with a specific `Mode`, but it seems like it requires me to specify the `where` clause, but I don't have a userId to reference and I'd like to avoid running two queries.

The key I use when upserting the `User` (discordId) is different from the key I need to upsert the `Profile` with (userId).

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

enum Mode {
  Ranked
  Unranked
  Open
}

model User {
  id        String    @id @default(uuid()) @db.Uuid
  profiles  Profile[]
  discordId String    @unique
  uuid      String    @unique @db.Uuid
  username  String
  createdAt DateTime  @default(now())
}

model Profile {
  user   User   @relation(fields: [userId], references: [id])
  userId String @db.Uuid
  mode   Mode
  wins   Int    @default(0)
  losses Int    @default(0)

  @@unique([mode, userId])
}
```

My query looks something like this:
```typescript
const user = await prisma.user.upsert({
	where: {
		discordId: source.user.id,
	},
	update: {
		username,
		profiles: {
			upsert: {
				where: {
					mode_userId: {
						mode: Mode.Open,
						userId: '?????', // what goes here?
					},
				},
				create: {
					mode: Mode.Open,
				},
				update: {
					wins: {
						increment: 1,
					},
				},
			},
		},
	},
	create: {
		discordId: source.user.id,
		username: 'Username'
	},
});
```

Modifying the schema would be fine, I'm just looking to complete this in one query.

Thanks for your help and let me know if you need any more info.

**Comment by nurul3101** [comment, marked as answer]
Hey @matteopolak 👋 
Apologies for the late reply, this GitHub Discussion slipped through the cracks.

I think creating a unique constraint on the combination of `userId` and `mode` would be ideal.
For the upsert query, a record needs to be uniquely identified in order to decide whether a record needs to be inserted or updated. 

In this case, as the uniqueness is dependent on both the userId and mode Prisma client prompts you to pass both the userId and mode.
Can you define the id of the parent document that needs to be inserted beforehand and pass that in the upsert statement?

**Reply by matteopolak** [reply]
No worries!

The userId field is an auto-generated uuid so I can't specify it. I've restructured everything since I submitted this as I believe this is an impossible thing to do.

I guess the only way (if I want to use a generated uuid) is to create it myself and then use it, right?

**Reply by nurul3101** [reply]
Yes, that would be correct.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
